java-checks/src/test/files/checks/RedundantTypeCastCheck.java

```diff
@@ -18,16 +18,16 @@
     Object o1 = (List<String>) foo(); // Noncompliant [[sc=18;ec=30]] {{Remove this unnecessary cast to "List".}}
     Object o2 = (List<? extends String>) foo(); // Noncompliant {{Remove this unnecessary cast to "List".}}
     Object o3 = (List<? super String>) foo(); // Noncompliant {{Remove this unnecessary cast to "List".}}
-    String s1 = (String) obj; //Compliant
+    String s1 = (String) obj; // Compliant
     String s2 = (String) s1; // Noncompliant {{Remove this unnecessary cast to "String".}}
     A a = (A) new B(); // Noncompliant {{Remove this unnecessary cast to "A".}}
     A[][] as = (A[][]) new B[1][1]; // Noncompliant {{Remove this unnecessary cast to "A[][]".}}
     B b;
     fun(b);
-    fun((A) b);
+    fun((A) b); // Compliant - exception to distinguish the method to call
     List<B> bees = new java.util.ArrayList<B>();
     java.util.List<A> aaas = (java.util.List) bees;
-    C c = new C((A)null);
+    C c = new C((A)null); // Compliant - exception to distinguish the constructor to call
```

Comment:
I would add a new constructor call to check that we are new raising issues in newClassTree as well.